Clients/GUI/Shared/Header.razor

```diff
@@ -1,38 +1,90 @@
+        <div class="collapse navbar-collapse" id="navbarNav">
+            <ul class="navbar-nav mr-auto">
+                <li class="nav-item">
+                    <a class="nav-link" href="/">News</a>
+                </li>
+                <AuthorizeView>
+                    <Authorized>
+                        <li class="nav-item">
+                            <a class="nav-link" href="/instruments">Instruments</a>
+                        </li>
+                    </Authorized>
+                </AuthorizeView>
+            </ul>
+            <div class="navbar-nav ml-auto">
+                <AuthorizeView>
+                    <NotAuthorized Context="NotAuth">
+                        <button type="button" class="btn btn-outline-secondary mr-2" @onclick="@(() => modal.Show<GUI.Pages.SignIn>())">Sign In</button>
+
+                        <button type="button" class="btn btn-outline-secondary" @onclick="@(() => modal.Show<GUI.Pages.SignUpWizzard>())">Sign up</button>
+                    </NotAuthorized>
+                    <Authorized>
+                        <div class="btn-group dropleft">
+                            <a class="navbar-brand" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                                <img src="@imgSrc" @onload="LoadImageSrc" width="30" height="30" class="d-inline-block align-top rounded-circle" alt="Avatar">
+                            </a>
+
+                            <div class="dropdown-menu">
+                                <a class="dropdown-item">
+                                <img src="@imgSrc" @onload="LoadImageSrc" width="30" height="30" class="align-items-center rounded-circle" alt="Avatar">
+                                    </a>
+                                <a class="dropdown-item" href="userinfo">User Info</a>
+                                <div class="dropdown-divider"></div>
+                                <a class="dropdown-item" href="/" @onclick="SignOut">Sign out</a>
+                            </div>
+                        </div>
+                    </Authorized>
+                </AuthorizeView>
+            </div>
+        </div>
+    </nav>
 </div>
 
 @code {
+    private string imgSrc = "Picture/Avatar.svg";
+
     private void SignOut()
     {
         Storage.ClearAsync();
+
         AuthenticationProvider.MarkSignedOut();
     }
-}
 
+    private async void LoadImageSrc()
+    {
+        var userToken = new UserToken
+        {
+            Body = await Storage.GetItemAsync<string>("token"),
+            UserId = await Storage.GetItemAsync<Guid>("id")
+        };
+
+        if (userToken.Body == null)
+        {
+            return;
+        }
+
+        var requestResult = await UserGetter.GetUserById(userToken);
+
+        if (requestResult.Avatar != null)
+            imgSrc = GetUserImage(requestResult.Avatar, requestResult.AvatarExtension);
+    }
+
+    private string GetUserImage(byte[] avatar, string extension)
```

Comment:
Why don't we achieve just userAvatar model?

Reply:
added